src/Sign.Core/ServiceProviderFactory.cs

```diff
@@ -9,12 +9,27 @@ namespace Sign.Core
 {
     internal sealed class ServiceProviderFactory : IServiceProviderFactory
     {
+        private Action<IServiceCollection>? _servicesToAdd;
+
         public IServiceProvider Create(
             LogLevel logLevel = LogLevel.Information,
             ILoggerProvider? loggerProvider = null,
             Action<IServiceCollection>? addServices = null)
         {
+
+            if (_servicesToAdd is not null)
+            {
+                addServices += _servicesToAdd;
+            }
+
             return ServiceProvider.CreateDefault(logLevel, loggerProvider, addServices);
         }
+
+        public void AddServices(Action<IServiceCollection> services)
+        {
+            ArgumentNullException.ThrowIfNull(services, nameof(services));
+
+            _servicesToAdd += services;
```

Comment:
```suggestion
        private Action<IServiceCollection>? _addServices;

        public IServiceProvider Create(
            LogLevel logLevel = LogLevel.Information,
            ILoggerProvider? loggerProvider = null,
            Action<IServiceCollection>? addServices = null)
        {
            if (_addServices is not null)
            {
                addServices += _addServices;
            }

            return ServiceProvider.CreateDefault(logLevel, loggerProvider, addServices);
        }

        public void AddServices(Action<IServiceCollection> addServices)
        {
            ArgumentNullException.ThrowIfNull(addServices, nameof(addServices));

            _addServices += addServices;
```

Delegates, like methods, are an action and should convey action.